[WOOMOB-971] Fix first order not auto-selecting on tablet when navigating from dashboard

kidinov opened the pull request:
WOOMOB-971

### Description

Fixed an issue where tapping "View All Orders" on the Recent Orders dashboard card doesn't auto-select the first order on tablets. 

The bug was apparently caused by fix of the another bug

```
                    communicationViewModel.event.value is
                    OrdersCommunicationViewModel.CommunicationEvent.OrdersLoaded -> {
                        // Prevents unintended navigation issues when opening an order list/detail in tablets.
                        // When navigating from order creation to order details via the "Collect Payment" option,
                        // the app correctly opens the Select Payment fragment inside the order details flow.
                        // However, if the above condition is not present, this navigation is undone,
                        // and only the order details screen is shown (skipping the Select Payment fragment).
                        // This no-op block ensures the app doesn't mistakenly re-trigger the order details screen.
                    }
```

The solution is to not match this block if there is selected order id

### Testing information

1. Open the app on a tablet device or emulator
2. Navigate to My Store and tap "View All Orders" on the Recent Orders card  
3. Verify the first order is automatically selected and displayed in the detail pane
4. Test after manually selecting orders to ensure it still works consistently

### The tests that have been performed

- Verified first order auto-selects when navigating from dashboard on tablet
- Tested that the fix works even after previous manual order selections  
- Confirmed the original order creation → payment flow protection is preserved
- Verified existing navigation behavior remains unchanged


https://github.com/user-attachments/assets/02e6f698-b071-42bc-a92c-c94b359cddde

kidinov committed:
Fix logic for opening the first order when filtering is active

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

kidinov committed:
Fix first order auto-selection on tablet when navigating from dashboard

Pass orderId=0 as a special indicator to select the first order when navigating from dashboard's 'View All Orders' button to the order list on tablets

kidinov committed:
Refactor to use clearer navigation pattern and fix condition priority

- Replace magic number (0) with well-named constant SELECT_FIRST_ORDER_ID
- Fix condition order to ensure SELECT_FIRST_ORDER_ID is checked first,
  preventing it from being blocked by OrdersLoaded communication event
- This ensures first order selection works even after previous manual selections

kidinov committed:
Updated release notes

kidinov committed:
Merge branch 'trunk' into woomob-971-tapping-on-view-all-orders-on-the-recent-orders-dashboard

kidinov committed:
Refine OrdersLoaded condition to be more specific

Instead of completely bypassing OrdersLoaded logic, make it more specific:
- Only applies when orderId == -1L (no specific order context)
- This preserves the original order creation → payment flow fix
- While allowing dashboard navigation with SELECT_FIRST_ORDER_ID to work properly

kidinov committed:
Simplify fix to use existing default case logic

Remove the special ID approach entirely and instead just make the OrdersLoaded
condition more specific by adding '&& viewModel.orderId.value == -1L'. This allows
the existing default case to handle first order selection properly while preserving
the original order creation → payment flow fix.

kidinov committed:
Reduce handler delay from 300ms to 200ms for improved responsiveness

kidinov committed:
Removed redundant comment

kidinov committed:
Add back special navigation ID to fix dashboard navigation

When not passing any orderId from dashboard, it defaults to -1L which still gets
blocked by the OrdersLoaded condition. Using SELECT_FIRST_ORDER_ID constant to
explicitly indicate dashboard navigation and auto-select first order on tablets.

kidinov committed:
Refactor order selection logic to improve clarity and maintainability

kidinov committed:
Fixed formatting

samiuelson reviewed (commented):
Despite the changes you made, the issue is still present on my device/store. Take a look:
<video src=https://github.com/user-attachments/assets/31fde6bb-26c5-4d00-beb9-88722b7f3a77/>

I tried debugging the problem:
1. `openFirstOrder(it)` is called in OrderListFragment:570
2. `if (firstOrder.orderId != selectedOrder.selectedOrderId.value)` body is not executed (condition is false) in OrderListFragment:859
This suggests that `SelectedOrderTrackerViewModel` holds the correct `selectedOrderId` value, but the order is not correctly rendered in UI for some reason 🤔

kidinov commented:
@samiuelson 

Something with the video. Could you please upload it again? 

Also, make sure that you test without debug attached

kidinov committed:
Fixed release notes

kidinov committed:
Remove unused constant for first order ID and simplify order navigation logic

kidinov committed:
Remove unused import statement in DashboardOrdersCard.kt

kidinov committed:
Merge branch 'trunk' into woomob-971-tapping-on-view-all-orders-on-the-recent-orders-dashboard

kidinov commented:
@samiuelson I fixed that, please review again!

kidinov committed:
Remove unnecessary delay when opening a specific order in OrderListFragment

samiuelson commented on RELEASE-NOTES.txt:
```diff
@@ -4,6 +4,7 @@
 
 23.1
 -----
+- [*] - [*] Fixed bug when an order was not selected when navigating from my store screen [https://github.com/woocommerce/woocommerce-android/pull/14436]
```
```suggestion
- [*] Fixed bug when an order was not selected when navigating from my store screen [https://github.com/woocommerce/woocommerce-android/pull/14436]
```

samiuelson reviewed (approved):
Tests good 👍

kidinov committed:
Update RELEASE-NOTES.txt

Co-authored-by: Samuel Urbanowicz <samiuelson@gmail.com>
